### PR TITLE
perf: optimize dockerfile

### PR DESCRIPTION
**Pull request by thebedigupta**
**Description**
In this PR I worked on this issues

Attaching updated test results file after the changes which I made [test-results.txt](https://github.com/user-attachments/files/17265693/test-results.txt)

1. [Remove cache after installing packages or store it in a cache mount](https://sonarcloud.io/project/issues?issueStatuses=OPEN%2CCONFIRMED&id=asyncapi_generator&open=AZB4sekEgviDxNeZYbKm)
2. [Merge this RUN instruction with the consecutive ones](https://sonarcloud.io/project/issues?issueStatuses=OPEN%2CCONFIRMED&id=asyncapi_generator&open=AZFCmY8SYFR-KciwR6Ur)
3. [Surround this variable with double quotes; otherwise, it can lead to unexpected behavior](https://sonarcloud.io/project/issues?issueStatuses=OPEN%2CCONFIRMED&id=asyncapi_generator&open=AZB4sekEgviDxNeZYbKn)
4. [Include the ARG instruction in the build stage where it is used](https://sonarcloud.io/project/issues?issueStatuses=OPEN%2CCONFIRMED&id=asyncapi_generator&open=AZB4sekEgviDxNeZYbKo)

**Results**
- avoids caching the index, which reduces the image size.
- decrease the no of layering in the docker image.

**I solved this issue after reading sonarcloud suggestion and how it work**

**Related issue(s)**

### CODE REVIEW

File: apps/generator/cli.js
Comment (DhairyaMajmudar) [inline apps/generator/cli.js:1]: Pls. Don't delete this file, get it back

Comment (thebedigupta) [reply apps/generator/cli.js:1]: done

**Review by derberg — COMMENTED** [review]
please remove `cli.js` changes

**Comment by thebedigupta** [comment]
> please remove `cli.js` changes

done

**Review by derberg — COMMENTED** [review]
`cli.js` is still showing up in the list of modified files

**Comment by DhairyaMajmudar** [comment]
Pls. have a look on the comment https://github.com/asyncapi/generator/pull/1283#issuecomment-2404066707 to fix this

**Pull request closed without merging**